### PR TITLE
Add deploy scripts

### PR DESCRIPTION
**Pull request by jharrington22**
This PR adds the required scripts to deploy this operator via the APP-SRE pipeline.

### CODE REVIEW

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    displayName: aws-account-operator Registry
+    publisher: SRE 
+
+- apiVersion: operators.coreos.com/v1alpha2
+  kind: OperatorGroup
+  metadata:
+    name: aws-account-operator-og
+  spec:
+    targetNamespaces:
+    - hive
```
Comment (nimrodshn) [inline hack/olm-registry/olm-artifacts-template.yaml]: Should this be the same namespace as `hive`? maybe `aws-account-operator`?

Comment (jharrington22) [reply hack/olm-registry/olm-artifacts-template.yaml]: Fixed!

**Comment by jharrington22** [comment]
@nimrodshn @jmelis CR please! :)

cc @fahlmant @rafael-azevedo

**Comment by nimrodshn** [comment]
@jharrington22 LGTM :+1: :100:

File: hack/app_sre_create_image_catalog.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -exv
+
+BRANCH_CHANNEL="$1"
+QUAY_IMAGE="$2"
+
+GIT_HASH=$(git rev-parse --short=7 HEAD)
+GIT_COMMIT_COUNT=$(git rev-list 9a5e025f954009144cdc403acbdc3cb5178addf5..HEAD --count)
```
Comment (jmelis) [inline hack/app_sre_create_image_catalog.sh]: in the interest of being able to reuse this, I would replace this with:

    GIT_COMMIT_COUNT=$(git rev-list $(git rev-list --max-parents=0 HEAD)..HEAD --count)

File: hack/app_sre_create_image_catalog.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        [ -d "$BUNDLE_DIR/$version" ] || continue
+
+        if [[ "$delete" == false ]]; then
+            short_hash=$(echo "$version" | cut -d- -f2)
+
+            if [[ "$DEPLOYED_HASH" == "${short_hash}"* ]]; then
+                delete=true
+            fi
+        else
+            rm -rf "${BUNDLE_DIR:?BUNDLE_DIR var not set}/$version"
```
Comment (jmelis) [inline hack/app_sre_create_image_catalog.sh]: oh, lovely tip! didn't know this

File: hack/app_sre_create_image_catalog.sh (hunk 2 above)
Comment (jharrington22) [reply hack/app_sre_create_image_catalog.sh]: Nice one! I have to replicate this to 3 other repos so I plan to create a repo or something to centralize this and make it reusable.

**Comment by jmelis** [comment]
LGTM too

**Pull request merged**